chore: empire base renames and docs

just-mitch opened the pull request:
Lot of comments on EmpireBase, GovernanceProposer, and GSEPayload, and introduces some renames.

Generally, on the EmpireBase now, the IEmperor is used to determine the current "signaler", who may "signal" on a "payload"; I want to preserve "proposal" and "vote" for governance, i.e. when a payload is actually `propose`ed to Governance.sol, it becomes part of a "proposal". 

Further, the `_execute` virtual function is renamed to `_handleRoundWinner`, so you don't "executeProposal" on the EmpireBase anymore, instead you "submitRoundWinner" (which calls `_handleRoundWinner`).

Also, in round accounting, `lastVote` is now `lastSignalSlot`, and `leader` is `payloadWithMostSignals`.

just-mitch committed:
chore: empire base renames and docs

just-mitch commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/15750?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#15750** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/15750?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/15750?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#15678** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/15678?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#15294** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/15294?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

LHerskind commented on l1-contracts/src/governance/GSEPayload.sol:
```diff
@@ -10,15 +10,26 @@ import {IProposerPayload} from "./interfaces/IProposerPayload.sol";
+ *
+ * In such an event, your recourse is either:
+ * - wait for the canonical rollup to have at least 2/3 of the total stake
+ * - `GSE.proposeWithLock`, which bypasses the GovernaceProposer
```
```suggestion
 * - `GSE.proposeWithLock`, which bypasses the GovernanceProposer
```

LHerskind commented on l1-contracts/src/governance/Governance.sol:
```diff
@@ -269,13 +269,18 @@ contract Governance is IGovernance {
     override(IGovernance)
     onlySelf
   {
+    require(
```
:+1:

LHerskind commented on l1-contracts/src/governance/proposer/GovernanceProposer.sol:
```diff
@@ -12,16 +12,25 @@ import {IRegistry} from "@aztec/governance/interfaces/IRegistry.sol";
+ *
+ * Note: any payload which passes through this contract will have a call to GSEPayload.amIValid appended to the
+ * list of actions before it is proposed to Governance.
+ * This will cause the proposal to revert if 2/3 of all stake in the GSE are not staked on the canonical rollup after
```
```suggestion
 * This will cause the proposal to revert if >2/3 of all stake in the GSE are not staked on the canonical rollup after
```

just-mitch committed:
chore: GSE docs and renames (#15810)

Adds clarifying comments/renames primarily to GSE and AddressSnapshotLib

The big change is being very explicit about what "canonical" means,
which is reserved for "the latest rollup according to the registry".

The GSE has its own, separate view of "rollups" that have been added to
it.

"Rollups" are "instances" according to the GSE, and there is a special
"bonus" instance that is available for use by the most recently added
rollup.
